include/glow/Partitioner/Partitioner.h

```diff
@@ -295,17 +298,26 @@ class Partitioner {
   /// "Function Family", that is, without considerting the "dynamic stuff" (i.e.
   /// batch size, input/output shape of each op), all the functions are
   /// identical. The required memory and computation cost for each op can be
-  /// found in Module. The \p devices provides the cost model related to
-  /// devices.
+  /// found in Module.
+  /// The \p devices provides the cost model related to devices.
+  /// Saturating the host will be enabled if \p saturateHost is true.
+  /// \p optimized is false by default, which means the functions in this module
+  /// are not optimized. \p partitionConfig contains the user defined partition
+  /// info.
   Partitioner(Module *parent, const std::vector<DeviceInfo> &devices,
-              bool saturateHost = false, bool optimized = false);
+              bool saturateHost = false, bool optimized = false,
+              PartitionConfig partitionConfig = PartitionConfig());
```

Comment:
I have a question about API. Why do we have to change the constructor, this common API used by everyone? Why can't we leave it as is, but pass `PartitionConfig` into `PartitionFromConfig()` method? It is a bit obscure why we accept `PartitionConfig` here, store it, but delay using it until `PartitionFromConfig()` is called.

Reply:
The reason is, we would like to leave the decision to Partitioner, and from all caller's the Partitioner interface is the same.  Later, we are going to refactor the Partitioner. After that, the Partitioner will be a facade that dispatches to the desired subclass/method.

Reply:
> from all caller's the Partitioner interface is the same

Why do we need this? This cannot be achieved. Imagine that one caller has user-defined partitions struct, the other caller doesn't. It means that caller #1 must pass this struct to partitioner, the second caller has nothing to pass. In caller Caller #1 calls `PartitionFromConfig()`, caller #2 calls something else. So interface is already different.

Reply:
The interface for both caller 1 and 2 is Partition(). The partitionConfig is passed when the Partitioner is initialed. So in Partition(), it will check if "partitionConfig" is enabled, if so, PartitionFromConfig() will be called. Otherwise, just continue the auto-partition.

Reply:
You can see that it's already bad API. Let's say caller 1 has only `partitionConfig`. But now it has to pass `false, false` for some other 2 flags. Caller 1 has to figure out what is saturation, and what to pass for corresponding flag to disable.

It would be a good practice to make sure that incompatible flags are not passed together. E.g. saturation=false when user defined partition. But these checks can quickly explode if we have more flags/modes.

What if we later introduce another feature, third mode for partitioner? Say, PyTorch-defined partition? Then caller 3 will have to figure out what to pass for partitionConfig argument to disable user-defined partition feature.

This is why I think we should never accept a whole all-inclusive list of args in one function (in this case, partitioner constructor). I was going to propose create different method for user defined partition. But if you are going to refactor everything later into different classes, I'm also fine with that.